vsphere/test_vsphere.py

```diff
@@ -335,3 +336,102 @@ def test_mor_discovery(self):
                 u"vsphere_host:host3", u"vsphere_type:vm"
             ]
         )
+
+
+class TestVsphereTopo(AgentCheckTest):
+
+    CHECK_NAME = "vsphere"
+
+    def vm_mock_content(self):
+        datastore = MockedMOR(_moId="54183927-04f91918-a72a-6805ca147c55")
+        hardware = MockedMOR(numCPU=1, memoryMB=4096)
+        config = MockedMOR(guestId='ubuntu64Guest', guestFullName='Ubuntu Linux (64-bit)', hardware=hardware)
+        view_MOR = MockedMOR(spec="VirtualMachine", name="Ubuntu",
+                             datastore=[datastore], config=config)
+
+        view_mock = MagicMock(view=[view_MOR])
+        viewmanager_mock = MagicMock(**{'CreateContainerView.return_value': view_mock})
+        content_mock = MagicMock(viewManager=viewmanager_mock)
+        return content_mock
+
+    def test_vsphere_objs(self):
+        """
+        Test if the vsphere_objs returns the VM list
+        """
+        config = {}
+        self.load_check(config)
+        self.check._is_excluded = MagicMock()
+        self.check._is_excluded.return_value = False
+
+        content_mock = self.vm_mock_content()
+        obj_list = self.check._vsphere_objs(content_mock, "vm", "ESXi")
+
+        self.assertEqual(len(obj_list), 1)
+        self.assertEqual(obj_list[0]['hostname'], 'Ubuntu')
+        # Check if labels are added
+        self.assertTrue(obj_list[0]['topo_tags']["labels"])
```

Comment:
Can we verify whether the resulted tags and labels are as expected, its contents?

Reply:
Added the assertion for both labels and tags

Reply:
Interesting.. Maybe the code path was unused before.